refactor(theme-default): update table styling for v2

mrrobot2410 opened the pull request:
## Summary
Internal refactor updating table styling in the default theme as part of the v2 default-theme table improvements.

## Changes
- Update table styles in default theme for v2

---

<details>
<summary>🇩🇪 Deutsche Zusammenfassung</summary>

## Zusammenfassung
Interner Refactor: Tabellen-Styles im Default-Theme als Teil der v2-Default-Theme-Tabellenverbesserungen aktualisiert.

## Änderungen
- Tabellen-Styles im Default-Theme für v2 aktualisiert

</details>

mweiershaeuser committed:
feat: add table styles

Refs: #7817

mweiershaeuser committed:
feat: added header checkbox styling

Refs: #7817

mweiershaeuser committed:
Merge remote-tracking branch 'origin/release/2' into 7817-default-theme-tabelle-2-v2

mweiershaeuser committed:
feat: adjust box-shadow for header checkbox

Refs: #7817

mweiershaeuser committed:
feat: add background colour to header td

Refs: #7817

deleonio commented:
Hi @mrrobot2410,

leider ist jetzt ein kleiner Konflikt drin. Könnt Ihr das bitte nach Euren Vorstellungen fixen. Danke

deleonio committed:
Update deps and lock

deleonio committed:
fix(a11y.scss): replace deprecated clip property with clip-path for visually hidden elements

deleonio committed:
Update component documentation for improved clarity and functionality

- Enhanced `_msg` property descriptions across multiple components to include updated structure and options.
- Added return descriptions for `getValue()` and `kolFocus()` methods in various components to clarify their functionality.
- Updated `closeModal()` and `openModal()` methods in the modal component to specify their effects on the modal dialog.
- Improved descriptions for `getSelection()` in the table component to clarify the data returned.
- Added functionality descriptions for tree item methods such as `collapse()`, `expand()`, and `focusLink()`.
- Clarified the behavior of `enqueue()` method in the toaster component to include the new `onClose` parameter.

deleonio committed:
fix(nav.scss): remove deprecated webkit-box-align property and adjust styles
chore(stylelintrc): add rule for nesting selector scoping

deleonio committed:
fix: update test-update scripts to use --update-snapshots=changed for improved snapshot management

deleonio committed:
fix: update message properties in component documentation

- Adjusted  property type in input-radio, input-range, input-text, select, single-select, textarea components to include default as a possible type.
- Added spacing and formatting improvements in various component readme files for better readability.
- Updated deprecated notice in logo component documentation.
- Enhanced descriptions and clarified attributes in modal, nav, pagination, popover-button, progress, and toaster components.

deleonio committed:
Update all snapshots

deleonio committed:
Update deps and lock in v2.2.14 (#8346)

mweiershaeuser committed:
Merge remote-tracking branch 'origin/release/2' into 7817-default-theme-tabelle-2-v2

Refs: #7817

mweiershaeuser commented:
@deleonio Wir haben jetzt den release/2 Branch in unseren Branch reingemerged und die Konflikte aufgelöst. Du hast den Zielbranch ja geändert, ich glaube der hinkt release/2 jetzt noch hinterher. Diesen können wir allerdings selbst nicht aktualisieren. :)

deleonio commented:
@mrrobot2410 bitte noch die Lint-Fehler beheben. Ggf. reicht `pnpm lint:stylelint --fix`

https://github.com/public-ui/kolibri/actions/runs/17299504487/job/49462717708?pr=8336

mweiershaeuser committed:
fix: linting

Refs: #7817

mweiershaeuser commented:
@deleonio Ist erledigt. :)
